smithy4s-client-localstack/src/main/scala/kinesis4cats/smithy4s/client/localstack/LocalstackKinesisClient.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats.smithy4s.client
+package localstack
+
+import cats.effect.Async
+import cats.effect.kernel.Resource
+import cats.effect.syntax.all._
+import cats.syntax.all._
+import com.amazonaws.kinesis._
+import org.http4s.client.Client
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import smithy4s.aws._
+import smithy4s.aws.http4s._
+import smithy4s.aws.kernel.AwsRegion
+
+import kinesis4cats.localstack.LocalstackConfig
+import kinesis4cats.logging.LogEncoder
+import kinesis4cats.smithy4s.client.KinesisClient
+import kinesis4cats.smithy4s.client.middleware._
+
+object LocalstackKinesisClient {
```

Comment:
The implementation here is almost identical to the KinesisClient, except it adds an additional middleware to proxy the request, as well as uses some mock credentials.